src/main/java/net/snowflake/ingest/streaming/internal/ChannelCache.java

```diff
@@ -44,13 +67,73 @@ void addChannel(SnowflakeStreamingIngestChannelInternal<T> channel) {
   }
 
   /**
-   * Returns an iterator over the (table, channels) in this map.
+   * Get the last flush time for a table
+   *
+   * @param fullyQualifiedTableName fully qualified table name
+   * @return last flush time in milliseconds
+   */
+  Long getLastFlushTime(String fullyQualifiedTableName) {
+    FlushInfo tableFlushInfo = this.tableFlushInfo.get(fullyQualifiedTableName);
+    if (tableFlushInfo == null) {
+      throw new SFException(
+          ErrorCode.INTERNAL_ERROR,
+          String.format("Last flush time for table %s not found", fullyQualifiedTableName));
+    }
+    return tableFlushInfo.lastFlushTime;
```

Comment:
Do we need to guarantee synchronization between get and set flush time?

Reply:
Yes.The results of `get` will not be modified from other threads as we are creating new `FlushInfo` every time when calling `set`.